### PR TITLE
dm: vga: remove all the assertion.

### PR DESCRIPTION
**Pull request by psun3x**
To avoid assertions in devicemodel, remove all the assert() in vga.c, use print error message to check errors.

Tracked-On: #8125
Signed-off-by: Sun Peng <peng.p.sun@linux.intel.com>
Reviewed-by: Huang, Yonghua <yonghua.huang@intel.com>
Acked-by: Wang, Yu1 <yu1.wang@intel.com>

### CODE REVIEW

File: devicemodel/hw/vga.c [hunk 1]
```diff
@@ -1258,7 +1243,9 @@ vga_init(struct gfx_ctx *gc, int io_only)
 		iop.arg = vd;
 
 		error = register_inout(&iop);
-		assert(error == 0);
+		if (error) {
+			pr_err("%s: regitster_inout error %d.\n", __func__, error);
```
Comment (ywan170) [inline devicemodel/hw/vga.c:1247]: The return value needs to be defined.

File: devicemodel/hw/vga.c [hunk 2]
```diff
@@ -1274,7 +1261,9 @@ vga_init(struct gfx_ctx *gc, int io_only)
 	vd->mr.handler = vga_mem_handler;
 	vd->mr.arg1 = vd;
 	error = register_mem_fallback(&vd->mr);
-	assert(error == 0);
+	if (error) {
+		pr_err("%s: regitster_mem_fallback error %d.\n", __func__, error);
+	}
```
Comment (ywan170) [inline devicemodel/hw/vga.c:1266]: Ditto

**Comment by szhen11** [comment]
@ywan170 Please help to review the force-pushed updates from psun3x.

**Review by yakuizhao — APPROVED** [review]
LGTM

**Pull request merged**